dwarf/line/state_machine.go

```diff
@@ -151,9 +161,10 @@ func execSpecialOpcode(sm *StateMachine, instr byte) {
 
 	sm.lastDelta = int(sm.dbl.Prologue.LineBase + int8(decoded%sm.dbl.Prologue.LineRange))
 	sm.line += sm.lastDelta
-	sm.address += uint64(decoded / sm.dbl.Prologue.LineRange)
+	sm.address += uint64(decoded/sm.dbl.Prologue.LineRange) * uint64(sm.dbl.Prologue.MinInstrLength)
```

Comment:
I'm doubtful about this change.  I ended up with this, instead:
```
                sm.isStmt = true
        }
 
-       sm.lastDelta = int(sm.dbl.Prologue.LineBase + int8(decoded%sm.dbl.Prologue.LineRange))
+       var deltapc uint64
+       var deltaline int
+
+       switch opcode {
+       case 249:
+               deltaline = 5
+               deltapc = 23
+       case 255:
+               deltaline = 1
+               deltapc = 24
+       default:
+               deltaline = int(sm.dbl.Prologue.LineBase + int8(decoded%sm.dbl.Prologue.LineRange))
+               deltapc = uint64(decoded / sm.dbl.Prologue.LineRange)
+       }
+
+       sm.lastDelta = deltaline
        sm.line += sm.lastDelta
-       sm.address += uint64(decoded / sm.dbl.Prologue.LineRange)
+       sm.address += deltapc
        sm.basicBlock = false
        sm.lastWasStandard = false
```
I got this from https://go.googlesource.com/go/+/f9feaffdf51967bc5cf3c9363db9ddee98c9b3a0/src/cmd/link/internal/ld/dwarf.go#918

```
		} else {
			// Use opcode 249 (pc+=23, lc+=5) or 255 (pc+=24, lc+=1).
			//
```
and I don't see any way that the change proposed here takes us from a state where the PC increments were not 23 and 24 to one where they are (because 23 and 24 have no common factors other than 1).